dali/sasha/sadbghk.cpp

```diff
@@ -0,0 +1,224 @@
+        CSashaSchedule schedule;
+        if (interval)
+            schedule.init(props, interval, initinterval);
+        initinterval *= 60 * 60 * 1000; // ms
+        unsigned started = msTick();
+        while (!stopped)
+        {
+            stopsem.wait(1000 * 60);
+            if (stopped)
+                break;
+            if (!interval || ((started != (unsigned)-1) && (msTick() - started < initinterval)))
+                continue;
+            started = (unsigned)-1;
+            if (!schedule.ready())
+                continue;
+            try
+            {
+                runDebugHousekeeping();
+            }
+            catch (IException *e)
+            {
+                StringBuffer s;
+                EXCLOG(e, LOGDBGHK);
+                e->Release();
+            }
+        }
+        PROGLOG(LOGDBGHK "Exit");
+        return 0;
+    }
+
+private:
+    bool isPostMortemDirPath(const StringBuffer &dirName)
+    {
+        // Expecting a directory name like "W20250225-101112"
+        RegExpr postMortemDirRegEx("^W[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]-[0-9][0-9][0-9][0-9][0-9][0-9]$");
+        if (postMortemDirRegEx.find(dirName.str()))
+        {
+            PROGLOG(LOGDBGHK "Post-mortem dir: %s", dirName.str());
+
+            return true;
+        }
+        else
+        {
+            PROGLOG(LOGDBGHK "Non post-mortem dir: %s", dirName.str());
+
+            return false;
+        }
+    }
+
+    bool isExpiredDirPath(StringBuffer &dirName, const unsigned &defaultExpireDays)
+    {
+        // Directory name is like "W20250225-101112"
+        StringBuffer cDateTimeoprmattedString;
+        cDateTimeoprmattedString.appendf("%c%c%c%c-%c%c-%c%cT%c%c:%c%c:%c%c",
+                                         dirName.charAt(1), dirName.charAt(2), dirName.charAt(3), dirName.charAt(4),
+                                         dirName.charAt(5), dirName.charAt(6),
+                                         dirName.charAt(7), dirName.charAt(8),
+                                         dirName.charAt(10), dirName.charAt(11),
+                                         dirName.charAt(12), dirName.charAt(13),
+                                         dirName.charAt(14), dirName.charAt(15));
+
+        CDateTime now;
+        now.setNow();
+
+        CDateTime expires;
+        expires.setString(cDateTimeoprmattedString.str());
+        expires.adjustTime(60 * 24 * defaultExpireDays);
+
+        if (now.compare(expires, false) > 0)
+        {
+            PROGLOG(LOGDBGHK "Post-mortem dir: %s has expired", dirName.str());
+
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+} *sashaDebugHousekeepingServer = NULL;
+
+ISashaServer *createSashaDebugHousekeepingServer()
+{
+    assertex(!sashaDebugHousekeepingServer); // initialization problem
+    Linked<IPropertyTree> config = serverConfig;
+    sashaDebugHousekeepingServer = new CSashaDebugHousekeepingServer(config);
```

Comment:
see other comments re. containerized, and using getComponentConfig*(). SO no need to pass in config here.
Worth adding a assertex(isContainerized() in this func for clarity.